### DISCUSSION TITLE
OPC-UA error (Bad Too Many Sessions)

### DISCUSSION THREAD

**Discussion by heritopo** [FreeOpcUa/opcua-asyncio#809] [Q&A]
Hello,
I am beginner about this topic.
I am trying to get multiple data from client (front cover station by FESTO) using python code. when i run the code, there was an error "Bad Too Many Session" as below picture.
![image](https://user-images.githubusercontent.com/96148109/154352384-19ed0840-4713-434d-b3f0-3af678c2a4bf.png)

server address : opc.tcp://10.48.134.10:4840
Signal address
![image](https://user-images.githubusercontent.com/96148109/154352170-0d5be9f8-b3ec-4a9b-be76-178cb7a98b4d.png)
![image](https://user-images.githubusercontent.com/96148109/154352961-fa31dfb9-bdec-4268-835b-532f3308f474.png)
picture of front cover station from festo 

Did the error appear due to code or server ?
Would you like to give solution regarding this ?

Thank you

**Comment by AndreasHeine** [comment]
BadToManySessions means that the OPC UA Server has no more resources for another Session!
Most Embedded small OPC UA Devices have just a few available, you need to check the manual for that device sometimes you can crank that session count up a little.

**Comment by heritopo** [comment]
Thank you for your explanation @AndreasHeine
I will check the manual and do it as your sugestion

**Comment by AndreasHeine** [comment]
@heritopo ?

**Comment by heritopo** [comment]
sorry @AndreasHeine, 
i can't find the literature of OPC UA server in the module,
do you have reference ?
and i realize that i should not make discussion in here. it supposed to be in discussion session.
if possible, I will transfer my issue in the discussion session and close this issue.
Is it ok ?
thanks

**Reply by AndreasHeine** [reply]
converted to a discussion ;)
